diff -ru mapnik-v3.1.0.orig/src/fs.cpp mapnik-v3.1.0/src/fs.cpp
--- mapnik-v3.1.0.orig/src/fs.cpp	2024-08-30 21:58:59.969635234 +0200
+++ mapnik-v3.1.0/src/fs.cpp	2024-08-30 22:00:48.488604302 +0200
@@ -29,6 +29,7 @@
 #include <mapnik/warning_ignore.hpp>
 #include <boost/filesystem/operations.hpp>  // for absolute, exists, etc
 #include <boost/filesystem/path.hpp>    // for path, operator/
+#include <boost/filesystem/directory.hpp>
 #pragma GCC diagnostic pop
 
 // stl
diff -ru mapnik-v3.1.0.orig/test/unit/imaging/image_io_test.cpp mapnik-v3.1.0/test/unit/imaging/image_io_test.cpp
--- mapnik-v3.1.0.orig/test/unit/imaging/image_io_test.cpp	2024-08-30 21:59:00.109638259 +0200
+++ mapnik-v3.1.0/test/unit/imaging/image_io_test.cpp	2024-08-30 22:10:55.207122334 +0200
@@ -15,7 +15,7 @@
 #pragma GCC diagnostic push
 #include <mapnik/warning_ignore.hpp>
 #include <boost/format.hpp>
-#include <boost/filesystem/convenience.hpp>
+#include <boost/filesystem/operations.hpp>
 #pragma GCC diagnostic pop
 
 inline void make_directory(std::string const& dir) {
diff -ru mapnik-v3.1.0.orig/test/unit/run.cpp mapnik-v3.1.0/test/unit/run.cpp
--- mapnik-v3.1.0.orig/test/unit/run.cpp	2024-08-30 21:59:00.109638259 +0200
+++ mapnik-v3.1.0/test/unit/run.cpp	2024-08-30 22:07:05.929456645 +0200
@@ -4,7 +4,7 @@
 #include <string>
 #include <mapnik/util/fs.hpp>
 #include <mapnik/datasource_cache.hpp>
-#include <boost/filesystem/convenience.hpp>
+#include <boost/filesystem/operations.hpp>
 
 #include "cleanup.hpp" // run_cleanup()
 
